notion-server/src/ethereum/mod.rs

```diff
@@ -403,21 +437,42 @@ impl ShardChain {
     }
 }
 
+#[derive(Default, PartialEq, Copy, Clone, Debug)]
+struct Bytes32 {
+    pub bytes: [u8; 32],
+}
+
+impl Bytes32 {
+    fn from(bytes: &[u8]) -> [u8; 32] {
```

Comment:
I think I'd prefer this as a `try_from`, so that we don't accidentally convert slices of the wrong length. Thoughts?

Reply:
I have restructured the 32-bytes conversion more generally, including a length check.